File: 0151-reverse-words-in-a-string/0151-reverse-words-in-a-string.cpp
class Solution {
public:
    string reverseWords(string s) {
        stack<string> st;

        string str="";

        for(auto ch:s){
            if(ch ==' '){
                if(!str.empty()){
                st.push(str);
                str="";
                }
            }
            else str=str+ch;
        }
        if(!str.empty()) st.push(str);
        string ans="";

        while(!st.empty()){
            ans=ans+st.top();
            st.pop();
            if (!st.empty()) {
                ans += " ";
            }
        }

        return ans;
    }
};

File: 0151-reverse-words-in-a-string/0151-reverse-words-in-a-string.java
import java.util.*;

class Solution {
    public String reverseWords(String s) {
        Stack<String> st = new Stack<>();
        String word = "";

        for (char ch : s.toCharArray()) {
            if (ch == ' ') {
                if (!word.isEmpty()) {
                    st.push(word);
                    word = "";
                }
            } else {
                word += ch;
            }
        }
        
        if (!word.isEmpty()) st.push(word); 

        String ans = "";
        while (!st.isEmpty()) {
            ans += st.pop();
            if (!st.isEmpty()) ans += " ";
        }

        return ans;
    }
}
